esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2691,10 +2691,14 @@ bool CFileSprayEx::onCopy(IEspContext &context, IEspCopy &req, IEspCopyResponse
         StringBuffer destFolder, destTitle, defaultFolder, defaultReplicateFolder;
         StringBuffer srcNodeGroup, destNodeGroup;
         bool bRoxie = false;
+        const char* srcDali = req.getSourceDali();
         const char* destNodeGroupReq = req.getDestGroup();
-        if(!destNodeGroupReq || !*destNodeGroupReq)
+        if(isEmptyString(destNodeGroupReq))
         {
-            getNodeGroupFromLFN(context, srcname, destNodeGroup);
+            if(isContainerized() && !isEmptyString(srcDali))
```

Comment:
If dali specified performing a copy in containerized... 
That should normally be forbidden, it is at least very unusual (implies containerized has been configured with allowForeign=true).

I don't think that was what @cloLN was testing when he hit the problem.
As you can see from the JIRA srcDali was blank (as you'd expect),
destGroup was also blank, and it's that that caused the problem.

I think the issue is that the file was ~remote, and getNodeGroupFromLFN isn't using wsdfs::lookup.
(anywhere that might lookup files from a remote source should now be using wsdfs::lookup)

@wangkx - can you check 'lookupLogicalName' in esp/smc/SMCLib/LogicFileWrapper.cpp, change to wsdfs::lookup and test ?

Reply:
I am adding the code to call the wsdfs::lookup if the srcname isRemote. I got the error "Remote storage 'test_remote_storage_name' not found" because, I think, I haven't set the remote storage 'test_remote_storage_name'. My plan is to test the remote copy between HPCCs on my local Ubuntu VM and on my local Docker Desktop. How should I configure my environment.xml and my values.yaml to test the remote copy?

Reply:
>if the srcname isRemote. 

you should not need the test. wsdfs::lookup, will fallback to legacy DFS if needed.

>I think, I haven't set the remote storage 'test_remote_storage_name'. 

you need to setup a <RemoteStorage> definition on the bare-metal side.
See: https://track.hpccsystems.com/browse/HPCC-27688 (also, looking at the 160x setup may be helpful).

> copy between HPCCs on my local Ubuntu VM and on my local Docker Desktop. Is it possible?

Yes, you can hit a local k8s hpcc running in DD.

Reach out to me on Teams if you need guidance.

Reply:
I copied the setup from 160x (will need your guidance on how to hit the local k8s later). I still got the same "Remote storage 'name' not found" error. In https://github.com/hpcc-systems/HPCC-Platform/pull/16123, I saw the changes on several xsl files (no esp related?). Not sure that ESP can reach out the setup.